R/weight.R

```diff
@@ -62,12 +124,115 @@ setMethod("weightVariables", "VariableCatalog", function (x) {
     ## Return the names/aliases
     if (length(vars)) {
         return(sort(vapply(index(vars), vget(namekey(x)), character(1),
-            USE.NAMES=FALSE)))
+            USE.NAMES = FALSE)))
     } else {
         return(c())
     }
 })
 
+#' @rdname weightVariables
+#' @export
+setMethod("weightVariables<-", "CrunchDataset", function (x, value) {
+    weightVariables(allVariables(x)) <- value
+    x <- refresh(x)
+    return(x)
+})
+
+#' @rdname weightVariables
+#' @export
+setMethod("weightVariables<-", "VariableCatalog", function (x, value) {
+    modifyWeightVariables(x, value, type = "append")
+})
+
+
+#' modifyWeightVariables
+#'
+#' Change which variables can be set as a dataset's weight.
+#'
+#' @param x a dataset or variable catalog
+#' @param vars Variables to add or remove  this can be a numeric Crunch variable,
+#' list of numeric Crunch variables or a character vector with the aliases of numeric Crunch variables.
+#' @param type a character string determining how the weightVariables
+#' will be modified:
+#' - `"append"` : add `vars` to the current weight variables
+#' - `"remove"` : remove `vars` from the current list of weight variables
+#' - `"replace"`: replace the current weight variables with `vars`
+#'
+#' @export
+modifyWeightVariables <- function (x, vars, type = "append") {
+    if (is.dataset(x)) {
+        x <- allVariables(x)
```

Comment:
A confusing thing about doing this is that now the `x` you're returning from the function isn't the dataset. See below; I'd suggest only taking dataset as the argument. I think it simplifies your logic, following the existing models of [hideVariables](https://github.com/Crunch-io/rcrunch/blob/master/R/hide-variables.R#L39) and [deleteVariables](https://github.com/Crunch-io/rcrunch/blob/master/R/variable-delete.R#L16)

Reply:
Resolved by removing variable catalog support